### ISSUE TITLE
Add local feed reader

### ISSUE THREAD

**Issue by aa9000095** [Reeywhaar/want-my-rss#23]
Is it possible to add the option to subscribe to the feed via a local rss feed (such as QuiteRSS)?

**Comment by Reeywhaar** [comment]
There is an option in preferences already:
![screenshot 2019-02-15 at 20 56 01](https://user-images.githubusercontent.com/884649/52874964-27754c00-3164-11e9-8b90-000cd6b04945.png)

**Comment by aa9000095** [comment]
It does not allow to subscribe through QuiteRSS (or any feed reader that is not part of the web browser), no matter what you put in the preferences.

**Comment by Reeywhaar** [comment]
Oh, I must be mistaken it with just any website based reader. Am I understand right that QuiteRss is standalone app, like Reeder for example? Yeah, need to think about it. Great if such apps could have some protocol url registered, like `quiterss:subscribe=http://example.com/feed`, It could have make everything easier

**Comment by aa9000095** [comment]
Yep, It's a standalone app but I couldn't find documentation about how to add a feed from an external web browser, however before firefox removed support for rss feed it was possible to subscribe directly by clicking the "subscribe" button on the RSS preview.

Maybe Firefox was creating an .xml file of the feed and opening it with QuiteRSS. Not sure if it's still possible with the current implementation of plugins in Firefox but that could be the best (or only) way to support standalone feed reader.

**Comment by Reeywhaar** [comment]
Yeah, there was way, but it was removed with rss support: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/2/Adding_feed_readers_to_Firefox also, it was a duty of an reader app to register itself. As for now, there is a way, to communicate with standalone app via [native messaging](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging) but it requires app to have [native manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests). WebExtensions being web extensions, a lot of security limitations here.

The other way as I said, is to have own registered protocol as I said above.

**Comment by whoizit** [comment]
we lost this mime types in browser, maybe possible return it?
```
application/rss+xml
application/atom+xml
application/rdf+xml
x-scheme-handler/feed
```

**Comment by bolocholo** [comment]
I can easily subscribe to my local RSS feeds client, Liferea, by clicking on the feed's URL link (on top of the RSS feed page created by this extension) and selecting "Liferea" in the dialog that Firefox presents (that also allows to save the file locally). I can customize this dialog in "Preferences > General > Applications" (look for the row with "feed" in the "Content Type" column). I use Firefox 65.0.2 in a fully updated Manjaro Linux box.

**Comment by aa9000095** [comment]
Unfortunately it doesn't work with QuiteRSS.
Maybe QuiteRSS was using an API that Firefox had put in place to subscribe to feed reader rather than just opening a .rss or .atom file with it.

**Comment by whoizit** [comment]
@bolocholo can't find _feed_ in the "Content Type" column on my Firefox 65.0.1 from Void Linux. I have Liferea too, but it's not works for me. I click on subscribe in rss page opened by this addon, but nothing happens

**Comment by Reeywhaar** [comment]
So, I've been looking on this issue and still solution is unclear:

* @whoizit, there is a way for extension to [handle protocols](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/protocol_handlers), but it's hardly relative to your idea, and the list of protocols is very limited.
* @bolocholo, I can't find `feed` *(though I don't have any standalone reader)* in Content-Type column neither. This type must be remained with profile between Firefox updates, or something. What's interesting is how is it actually works? Reader must be provided with feed url for subscription, but file you open is just a piece of rss data which can contain feed url, but it's not obligatory per spec. I thought that I can add special `external` reader type, that will be roughly similar to clicking on `<a href="rss data url" download="feed.rss" type="x-scheme-handler/feed"></a>` but not sure if it be any helpful at all.

**Comment by tasadar2** [comment]
For those still interested, I managed to set this up with my local feed application. As @Reeywhaar mentioned, the application needs to handle a url protocol. Many old rss apps register `rss` and/or `feed`.
I use `feed:%s` in this extension to open in a local application.

If the application doesn't register a protocol itself, but takes commandline arguments, you can add it manually, see [http://kb.mozillazine.org/Register_protocol](http://kb.mozillazine.org/Register_protocol).
And it looks like quietrss does support arguments passed from non-primary instances to the primary instance, [https://github.com/QuiteRSS/quiterss/blob/0e9b869423/src/application/mainapplication.cpp#L135](https://github.com/QuiteRSS/quiterss/blob/0e9b869423/src/application/mainapplication.cpp#L135)

Only trouble I had was some applications expect the raw address and this extension encodes it, [https://github.com/Reeywhaar/want-my-rss/blob/3f33c014c3/ts/storage.ts#L106](https://github.com/Reeywhaar/want-my-rss/blob/3f33c014c3/ts/storage.ts#L106). An option to not encode the url or just a second replacement key like `%url` would be convenient.

**Comment by aa9000095** [comment]
Any update on when this fix will be merged?

**Comment by beroal** [comment]
> I can easily subscribe to my local RSS feeds client, Liferea, by clicking on the feed's URL link (on top of the RSS feed page created by this extension) and selecting "Liferea" in the dialog that Firefox presents (that also allows to save the file locally). I can customize this dialog in "Preferences > General > Applications" (look for the row with "feed" in the "Content Type" column). I use Firefox 65.0.2 in a fully updated Manjaro Linux box.

I also use Liferea on ArchLinux. This method does not **really** work. Firefox downloads a feed file to a local file system, then Liferea shows a feed referring to this **local file**. The local file will not be updated when the feed is updated and will disappear after a reboot because it resides in `/tmp`.

**Comment by beroal** [comment]
> Only trouble I had was some applications expect the raw address and this extension encodes it, https://github.com/Reeywhaar/want-my-rss/blob/3f33c014c3/ts/storage.ts#L106. An option to not encode the url or just a second replacement key like `%url` would be convenient.

Thank you for the suggestion. It seems to me that the URI syntax does not allow two schemes, and non-encoded “URI” has two schemes. For example, `feed:http://example.com` has the `feed` and `http` schemes. So passing a non-encoded “URI” is fragile since any program through which it passes (Firefox, desktop environment) may reject the non-encoded “URI” as invalid. I suppose we need to fix feed readers. 😓
